src/Donations/index.tsx

```diff
@@ -103,6 +103,7 @@ function DonationInfo() {
   } = React.useContext(QueryParamContext);
 
   const [isMobile, setIsMobile] = React.useState(false);
+  const [showBackButton, setShowBackButton] = React.useState(null);
```

Comment:
This is because I if the value if set to false at any point of time, it means that the user came from an external link and so I don't want to display the back button. Setting false initially would prevent me from updating the value and setting true would display the back button. Hence, I've set null as the initial value.